src/code/z_play.c

```diff
@@ -1,81 +1,1545 @@
+                    lower = ((r * 2 + g * 4 + b) * 0xF) / 217;
+
+                    *(destI4++) = (upper << 4) | lower;
+                }
+            }
+            break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165658.s")
+        case 5: {
+            u8* destI5 = (u8*)destI;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_8016566C.s")
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    pixel = srcRgba16[i * rgba16Width + j];
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165690.s")
+                    pixel = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_801656A4.s")
+                    *(destI5++) = (((r * 2 + g * 4 + b) * 0xFF) / 217) & 0xF8;
+                }
+            }
+            break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165DB8.s")
+        case 8: {
+            u8* destI8 = (u8*)destI;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165DCC.s")
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    pixel = srcRgba16[i * rgba16Width + j];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165DF0.s")
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165E04.s")
+                    *(destI8++) = ((r * 2 + g * 4 + b) * 0xFF) / 217;
+                }
+            }
+            break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165E1C.s")
+        case 16: {
+            u16* destI16 = (u16*)destI;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165E7C.s")
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    *(destI16++) = srcRgba16[i * rgba16Width + j];
+                }
+            }
+            break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80165EC0.s")
+void Play_SetMotionBlurPriorityAlpha(u32 alpha) {
+    R_MOTION_BLUR_PRIORITY_ALPHA = alpha;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/D_801DFA18.s")
+void Play_EnableMotionBlurPriority(u32 alpha) {
+    R_MOTION_BLUR_PRIORITY_ALPHA = alpha;
+    R_MOTION_BLUR_PRIORITY_ENABLED = true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80166060.s")
+void Play_DisableMotionBlurPriority(void) {
+    R_MOTION_BLUR_PRIORITY_ENABLED = false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_801660B8.s")
+// Will take the photograph, but doesn't compress and save it
```

Comment:
I think and is more appropriate here, as the actual saving of the photo is with compression. (Technically this function would save the new photo in `gPictoPhotoI8`, it just doesn't compress and save it to saveContext, so will just get overwritten.
As for Photo, it is to differentiate from the actual `Pictograph` item.

Reply:
The item is the Pictograph Box, though.

Reply:
Tbh, I always forget it is called the `Pictograph Box`. Hmm, @engineer124 we talked a little about this in Discord, perhaps it would be better if the parameter functions/enums used `PictographBox`, while this photo related things in play would just use `Pictograph`?

Reply:
I vote for using `Pictograph` and `PictographBox` to differentiate them

Reply:
If I was looking at this code fresh and saw `pictograph`, I would immediately think of the item and the system as a whole. I don't think I would see `pictograph` as the photo only, excluding the item, whereas `pictoPhoto` is much clearer

Reply:
Posted in [discord](https://discord.com/channels/688807550715560050/873211000122384415/1029582125718065283), but I like the idea of using `PICTO` as a general system that each part shares, than differentiating them with `PHOTO` for the actual photo, `BOX` for the item, and `CAPTURE` for the capture (what is currently in z64snap.h)